### PR TITLE
SDKS-1843 Logging out user from one app should logout the user from another app

### PR DESCRIPTION
**Pull request by witrisna**
# JIRA Ticket
https://bugster.forgerock.org/jira/browse/SDKS-1843

# Description
Logging out user from one app should logout the user from another app when the broadcast is not enabled.

# Definition of Done Checklist:

- [X] Acceptance criteria is met.
- [X] All tasks listed in the user story have been completed.
- [X] Coded to standards.
- [X] Ensure backward compatibility.
- [X] API reference docs is updated.
- [X] Unit tests are written.
- [X] Integration tests are written.
- [X] e2e tests are written.
- [X] Functional spec is written/updated.
- [X] contains example code snippets.
- [X] Change log updated.
- [X] Documentation story is created and tracked.
- [X] Tech debts and remaining tasks are tracked in separated ticket(s).

### CODE REVIEW

File: forgerock-auth/src/test/java/org/forgerock/android/auth/FRUserMockTest.java [hunk 1]
```diff
@@ -405,7 +405,7 @@ static void setFinalStatic(Field field, Object newValue) throws Exception {
     @Test
     public void testLogout() throws InterruptedException, ExecutionException, IOException, JSONException, ParseException, AuthenticationRequiredException, ApiException {
         frUserHappyPath();
-        Config.getInstance().setSSOModel(mockBroadcastModel);
+        Config.getInstance().setSSOBroadcastModel(mockBroadcastModel);
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/FRUserMockTest.java:408]: I think we can move this to setup and remove the duplicates  

  @Before
    public void setUp() {
        when(mockBroadcastModel.isBroadcastEnabled()).thenReturn(true);
        Config.getInstance().setSSOBroadcastModel(mockBroadcastModel);
    }

File: forgerock-auth/src/main/java/org/forgerock/android/auth/Token.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2019 ForgeRock. All rights reserved.
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/Token.kt:2]: update copy right

File: forgerock-auth/src/main/java/org/forgerock/android/auth/Token.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+package org.forgerock.android.auth
+
+import java.io.Serializable
+import java.util.*
+
+/**
+ * Domain object to hold generic Token
+ */
+open class Token(val value: String) : Serializable {
+    override fun equals(o: Any?): Boolean {
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/Token.kt:16]: just change this o to other . the IDE is complaining about this

**Review by jeyanthanperiyasamy — CHANGES_REQUESTED** [review]
some minor feedbacks

**Review by spetrov — APPROVED** [review]
Looks good to me. I tested and confirmed that the issue has been fixed.

File: forgerock-auth/src/main/java/org/forgerock/android/auth/Token.kt (hunk 2 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/Token.kt]: Updated

File: forgerock-auth/src/test/java/org/forgerock/android/auth/FRUserMockTest.java (hunk 1 above)
Comment (witrisna) [reply forgerock-auth/src/test/java/org/forgerock/android/auth/FRUserMockTest.java]: Moved to setup

**Review by jeyanthanperiyasamy — APPROVED** [review]

**Review by spetrov — APPROVED** [review]

**Pull request merged**